src/functions/indGraphSparse.jl

```diff
@@ -0,0 +1,56 @@
+struct IndGraphSparse{T <: RealOrComplex} <: IndGraph
+  m::Int
+  n::Int
+  A::AbstractArray{T,2}
+  F::Base.SparseArrays.CHOLMOD.Factor{T} #LDL factorization
+end
+
+function IndGraphSparse(A::AbstractArray{T,2}) where {T <: RealOrComplex}
+  m, n = size(A)
+  K = [speye(n) A'; A -speye(m)]
+
+  F = LinAlg.ldltfact(K)
+  #normrows = vec(sqrt.(sum(abs2.(A), 2)))
+
+  return IndGraphSparse(m, n, A, F)
+end
+
+# is_convex(f::IndGraph) = true
+# is_set(f::IndGraph) = true
+# is_cone(f::IndGraph) = false
+
+function prox!{T <: RealOrComplex}(
+    x::AbstractArray{T, 1},
+    y::AbstractArray{T, 1},
+    f::IndGraphSparse,
+    c::AbstractArray{T, 1},
+    d::AbstractArray{T, 1})
+  res = [c + f.A' * d; zeros(f.m)]
+  xy = f.F \ res
+  x[:] = xy[1:f.n]
+  y[:] = xy[end - f.m + 1:end]
```

Comment:
for these lines there are a few options to allocate less (almost no) memory, for example something like letting `res, xy` be allocated vectors in IndGraphSparse, and then
```julia
res1 =  view(res,1:n) #This is basically a pointer to only the first part of res, the other part should always remain zero
At_mul_B!(res1, f.A' d) # This sets the first part in res
res1 += c

A_ldiv_B!(xy, f.F, res1)

x .= view(xy,1:f.n) 
y .= view(xy,end - f.m + 1:end)
```
I am not sure if the two lines allocate memory as you wrote them, sometimes the syntax `xy[range]` on the right side does that. 
There are other alternatives that might be equally good as the two last lines i wrote, maybe more or less efficient or readable, like
```julia
copy!(x, 1, xy, 1, n)     #Copy to x starting at 1, from xy starting at 1, n elements
copy!(y, 1, xy, f.n+1, m) #Copy to y starting at 1, from xy starting at f.n+1, m elements
```
or using the `@views` macro which means the lines you wrote can be written
```julia
@views x .= xy[1:f.n]
@views y .= xy[end - f.m + 1:end]
```
which will transform into the view example I gave. This will only allocate a "view", like a pointer which is a constant and small allocation, the copy! should be the most efficient however.